browser/brave_wallet/sources.gni

```diff
@@ -3,12 +3,18 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 
-brave_browser_wallet_sources = [
-  "//brave/browser/brave_wallet/brave_wallet_service_factory.cc",
-  "//brave/browser/brave_wallet/brave_wallet_service_factory.h",
-]
-
-brave_browser_wallet_deps = [
-  "//components/keyed_service/content:content",
-  "//extensions/browser",
-]
+brave_browser_wallet_sources = []
+
+brave_browser_wallet_deps = []
+
+if (!is_android) {
```

Comment:
The original PR missed checking if **brave_wallet_enabled** is enabled. So, I've tried to use **brave_wallet_enabled** guard though, it has caused an import loop as below,

```
ERROR //build/config/chrome_build.gni is part of an import loop.
See //brave/components/brave_wallet/buildflags/buildflags.gni:1:1: whence it was imported.
import("//build/config/features.gni")
^-----------------------------------
See //brave/browser/brave_wallet/sources.gni:6:1: whence it was imported.
import("//brave/components/brave_wallet/buildflags/buildflags.gni")
^-----------------------------------------------------------------
See //brave/browser/sources.gni:7:1: whence it was imported.
import("//brave/browser/brave_wallet/sources.gni")
^------------------------------------------------
See //brave/build/config/brave_build.gni:6:1: whence it was imported.
import("//brave/browser/sources.gni")
^-----------------------------------
See //build/config/chrome_build.gni:4:1: whence it was imported.
import("//brave/build/config/brave_build.gni")
^--------------------------------------------
See //build/config/compiler/compiler.gni:5:1: whence it was imported.
import("//build/config/chrome_build.gni")
^---------------------------------------
See //BUILD.gn:11:1: whence it was imported.
import("//build/config/compiler/compiler.gni")
^--------------------------------------------
```

So, I used **is_android** guard instead. I wonder if this use is fine here.

Reply:
`components/brave_wallet/buildflags/buildflags.gni:  brave_wallet_enabled = enable_extensions`

Which is different from `!is_android`

What is the import loop?

Reply:
I've been trying to find a way to use brave_wallet_enabled in browser/brave_wallet/sources.gni. But, it caused an import loop error when importing "//brave/components/brave_wallet/buildflags/buildflags.gni" in //brave/browser/brave_wallet/sources.gni.

**//build/config/features.gni** -> //build/config/chrome_build.gni -> //build/config/compiler/compiler.gni -> //build/config/chrome_build.gni -> //brave/build/config/brave_build.gni" -> //brave/browser/sources.gni -> //brave/browser/brave_wallet/sources.gni -> //brave/components/brave_wallet/buildflags/buildflags.gni -> **//build/config/features.gni**

Only way I can use now is to use "(!is_android && !is_ios) instead of **brave_wallet_enabled** or **enable_extensions**. But, as you told me it can cause an out of sync issue. So, it looks like I need to revert the original PR(https://github.com/brave/brave-core/pull/6869), then, I would re-upload the original PR when I find a solution to avoid the import loop error. WDYT?

Reply:
Yes I don't think we should have `!is_android && !is_ios` where `brave_wallet_enabled` should be used. I don't know how best to solve it without jumping in the code myself. but maybe @iefremov or @bridiver can advise.

Reply:
@Gyuyoung components/brave_wallet/buildflags/buildflags.gni should not import `//build/config/features.gni`

Reply:
@bbondy  Thanks for the comment. Let me ask them soon. If there is no a good way to use brave_wallet_enabled, let me revert the original PR.

Reply:
@bridiver and @bbondy: ok, http://crrev.com/c/2516059 was merged. So, I would like to generate a patch file for it at the moment, then we will be able to remove the patch file via the next Chromium upgrade. I will file an issue to track the patch file after merging this PR. WDYT?

Reply:
@bridiver I wonder how do you think about this PR now. PTAL.

Reply:
> I think @mkarolin can also approve

@mkarolin Could you take a look at this PR instead of @bridiver?